src/main/java/io/knowledgeassets/myskills/server/community/RecommendedCommunityResponse.java

```diff
@@ -0,0 +1,64 @@
+package io.knowledgeassets.myskills.server.community;
+
+import io.knowledgeassets.myskills.server.skill.SkillResponse;
+import io.knowledgeassets.myskills.server.user.UserSimpleResponse;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+import static io.knowledgeassets.myskills.server.util.ConversionUtils.*;
+
+@Data
+@EqualsAndHashCode(callSuper = true)
+@AllArgsConstructor
+public class RecommendedCommunityResponse extends CommunityResponse {
+
+	@ApiModelProperty("Flag indicating if a community is recommended.")
+	private final boolean recommended;
+
+	@Builder(builderMethodName = "recommendedCommunityResponseBuilder")
+	public RecommendedCommunityResponse(String id,
+										String title,
+										CommunityType type,
+										String description,
+										List<LinkResponse> links,
+										List<UserSimpleResponse> managers,
+										List<UserSimpleResponse> members,
+										List<SkillResponse> skills,
+										boolean recommended) {
+		super(id, title, type, description, links, managers, members, skills);
+		this.recommended = recommended;
+	}
+
+	public static RecommendedCommunityResponse of(RecommendedCommunity recommendedCommunity) {
+		return RecommendedCommunityResponse.recommendedCommunityResponseBuilder()
+				.id(recommendedCommunity.getCommunity().getId())
+				.title(recommendedCommunity.getCommunity().getTitle())
+				.type(recommendedCommunity.getCommunity().getType())
+				.description(recommendedCommunity.getCommunity().getDescription())
+				.links(convertLinkListToLinkResponseList(recommendedCommunity.getCommunity().getLinks()))
+				.managers(convertUserListToUserSimpleResponseList(recommendedCommunity.getCommunity().getManagers()))
+				.members(convertUserListToUserSimpleResponseList(recommendedCommunity.getCommunity().getMembers()))
```

Comment:
I've thought about this for some time and somehow it feels wrong to have embedded sub-resources which depend on the user's authentication. I don't want to be dogmatic but I would try to avoid that for the sake of easier understanding of APIs and how server and web app are integrated.

In this case the question is: Why do we need to embed the members? Is it really such a big benefit for the web app to receive them together with the community data? On the community details page the members are displayed as a separate section anyway. We could load the member list lazily via `GET /communities/{commId}/members` and render them independently of the general community data. What do you think?

Reply:
We need to embed the members because the web app uses the member list [to determine if a user is a member of the community](https://github.com/KnowledgeAssets/myskills-webapp/blob/master/src/app/communities/communities.component.ts#L106) or not (to display / hide `Join` button).
So there has to be another way to decide if a user is a member of a community.

Reply:
Yes, I got it. We shouldn't change that in this feature branch.